fetch.bs

```diff
@@ -6714,6 +6753,435 @@ agent's <a>CORS-preflight cache</a> for which there is a <a>cache entry match</a
+ <a>deferred-fetch control document</a>.
+
+ <li><p>If the <a data-lt="define an inherited policy for feature in container">inherited policy</a>
+ for "{{PermissionsPolicy/deferred-fetch}}", <var>container</var> and <var>originToNavigateTo</var>
+ is <code>Enabled</code>, and the <a>available deferred-fetch quota</a> for
+ <var>controlDocument</var> is equal or greater than
+ <a for="reserved deferred-fetch quota">normal quota</a>, then set <var>container</var>'s
+ <a>reserved deferred-fetch quota</a> to <a for="reserved deferred-fetch quota">normal quota</a> and
+ return.
+
+ <li>
+  <p>If all of the following conditions are true:
+
+  <ul class=brief>
+   <li><p><var>controlDocument</var>'s <a>node navigable</a> is a <a for=/>top-level traversable</a>
+
+   <li><p>The <a data-lt="define an inherited policy for feature in container">inherited policy</a>
+   for "{{PermissionsPolicy/deferred-fetch-minimal}}", <var>container</var> and
+   <var>originToNavigateTo</var> is <code>Enabled</code>
+
+   <li><p>The <a for=list>size</a> of <var>controlDocument</var>'s <a>node navigable</a>'s
+   <a>descendant navigables</a>, <a for=list data-lt=remove>removing</a> any <a for=/>navigable</a>
+   whose <a>navigable container</a>'s <a>reserved deferred-fetch quota</a> is not
+   <a for="reserved deferred-fetch quota">minimal quota</a>, is less than
+   <a>quota reserved for <code>deferred-fetch-minimal</code></a> /
+   <a for="reserved deferred-fetch quota">minimal quota</a>.
+  </ul>
+
+  <p>then set <var>container</var>'s <a>reserved deferred-fetch quota</a> to
+  <a for="reserved deferred-fetch quota">minimal quota</a>.
+</ol>
+</div>
+
+<div algorithm>
+<p>To <dfn export>potentially free deferred-fetch quota</dfn> for a <a for=/>document</a>
+<var>document</var>, if <var>document</var>'s <a>node navigable</a>'s <a>container document</a> is
+not null, and its <a for=Document>origin</a> is <a>same origin</a> with <var>document</var>, then
+set <var>document</var>'s <a>node navigable</a>'s <a>navigable container</a>'s
+<a>reserved deferred-fetch quota</a> to 0.
+
+<p class=note>This is called when a <a for=/>document</a> is created. It ensures that same-origin
+nested documents don't reserve quota, as they anyway share their parent quota. It can only be called
+upon document creation, as the <a for=Document>origin</a> of the <a for=/>document</a> is only known
+after redirects are handled.
+</div>
+
+<div algorithm>
+<p>To get the <dfn>deferred-fetch control document</dfn> of a <a for=/>document</a>
+<var>document</var>:
+
+<ol>
+ <li><p>If <var>document</var>' <a>node navigable</a>'s <a>container document</a> is null or a
+ <a for=/>document</a> whose <a for=Document>origin</a> is not <a>same origin</a> with
+ <var>document</var>, return <var>document</var>; Otherwise return the
+ <a>deferred-fetch control document</a> given <var>document</var>' <a>node navigable</a>'s
+ <a>container document</a>.
```

Comment:
```suggestion
 <li><p>If <var>document</var>' <a>node navigable</a>'s <a>container document</a> is null or a
 <a for=/>document</a> whose <a for=Document>origin</a> is not <a>same origin</a> with
 <var>document</var>, then return <var>document</var>; otherwise, return the
 <a>deferred-fetch control document</a> given <var>document</var>'s <a>node navigable</a>'s
 <a>container document</a>.
```